src/problem_highs.jl

```diff
@@ -458,7 +483,7 @@ function solve!(p::HiGHS_Prob)
             status = Highs_getScaledModelStatus(p)
             println("Resetting solver due to solver status $(status): Rebuilding full LP and pass to solver")
         end
-        _passLP_reset!(p) # TODO: Now resets to simplex, make interface to keep settings
+        _passLP_reset!(p)
         ret = _Highs_run_reset_clock!(p)
     end
 
```

Comment:
Change that improves the logic in solve!:
1. Only try different simplex scaling strategies if solver type is simplex.
2. Only try primal and dual simplex if solver type is simplex (could also try if barrier, but then the settings has to be preserved)

use:
solver = pointer(Vector{Cchar}(undef, kHighsMaximumStringLength))
Highs_getStringOptionValue(p, "solver", solver)
        if unsafe_string(solver) != "ipm"